src/main/java/ru/sbt/mipt/oop/SmartHome.java

```diff
@@ -14,11 +16,48 @@ public SmartHome(Collection<Room> rooms) {
         this.rooms = rooms;
     }
 
+    public void processDoorEvent(SensorEvent event) {
+        for (Room room : this.getRooms()) {
+            Door door = room.getDoor(event.getObjectId());
+            if (door != null) {
+                room.changeDoor(door.getId(), (event.getType() == DOOR_OPEN));
+                if (room.getName().equals("hall") && event.getType() == DOOR_CLOSED) {
```

Comment:
https://github.com/iagoshkov/smart-home-2020/blob/master/STUDENT_GUIDE.md#%D1%80%D0%B0%D0%B7%D0%B4%D0%B5%D0%BB%D1%8F%D0%B9%D1%82%D0%B5-%D0%B8%D0%B7%D0%BC%D0%B5%D0%BD%D0%B5%D0%BD%D0%B8%D0%B5-%D0%BE%D0%B1%D1%8A%D0%B5%D0%BA%D1%82%D0%BE%D0%B2-%D0%B8-%D1%81%D1%86%D0%B5%D0%BD%D0%B0%D1%80%D0%B8%D0%B8-%D1%83%D0%BC%D0%BD%D0%BE%D0%B3%D0%BE-%D0%B4%D0%BE%D0%BC%D0%B0